### PR TITLE
fix loss computation by using shifted logits and targets to create co…

### PR DESCRIPTION
**Pull request by YAMWD**
fix loss computation by using shifted logits and targets. 

issue link [https://github.com/karpathy/llm.c/issues/799](url)

### CODE REVIEW

**Comment by YAMWD** [comment]
It also needs modification on softmax. Now the loss is correct.

image

**Comment by YAMWD** [comment]
I noticed that such shifts are done in the dataloader so the original implementation should be fine.
